x-pack/solutions/observability/plugins/apm/public/components/shared/trace_waterfall/trace_warning.tsx

```diff
@@ -19,7 +20,35 @@ const FALLBACK_WARNING = i18n.translate(
 );
 
 export function TraceWarning({ children }: { children: React.ReactNode }) {
-  const { rootItem } = useTraceWaterfallContext();
+  const { traceState } = useTraceWaterfallContext();
 
-  return !rootItem ? <EuiCallOut color="warning" size="s" title={FALLBACK_WARNING} /> : children;
+  switch (traceState) {
+    case TraceDataState.Full:
+      return children;
+
+    case TraceDataState.Partial:
+      return (
+        <EuiFlexGroup direction="column">
+          <EuiFlexItem>
+            <EuiCallOut
+              data-test-subj="traceWarning"
+              color="warning"
+              size="s"
+              title={FALLBACK_WARNING}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem>{children}</EuiFlexItem>
+        </EuiFlexGroup>
+      );
+
+    case TraceDataState.Empty:
+      return (
+        <EuiCallOut
+          data-test-subj="traceWarning"
+          color="warning"
+          size="s"
+          title={FALLBACK_WARNING}
+        />
+      );
+  }
```

Comment:
```suggestion
  switch (traceState) {
    case TraceDataState.Partial:
      return (
        <EuiFlexGroup direction="column">
          <EuiFlexItem>
            <EuiCallOut
              data-test-subj="traceWarning"
              color="warning"
              size="s"
              title={FALLBACK_WARNING}
            />
          </EuiFlexItem>
          <EuiFlexItem>{children}</EuiFlexItem>
        </EuiFlexGroup>
      );

    case TraceDataState.Empty:
      return (
        <EuiCallOut
          data-test-subj="traceWarning"
          color="warning"
          size="s"
          title={FALLBACK_WARNING}
        />
      );
      default:
          return children;
  }
```

Reply:
Just as a note (I'll still do the suggested): having all the cases be explicit means that if a new state is added, it will fail the Check Types step, as TS will(or should if set to strict mode) infer a missing case and will error. The default will never error if adding a new state.